src/main/java/edu/ksu/canvas/attendance/submitter/CanvasAssignmentAssistant.java

```diff
@@ -113,23 +113,29 @@ public void editAssignmentInCanvas(Long courseId, AttendanceAssignment attendanc
     public void deleteAssignmentInCanvas(Long canvasCourseId, OauthToken oauthToken) throws AttendanceAssignmentException{
         List<AttendanceSection> sections = attendanceSectionService.getSectionByCanvasCourseId(canvasCourseId);
         if(CollectionUtils.isEmpty(sections)) {
-            LOG.warn("Sections empty");
+            LOG.warn("Non existent sections for course when deleting assignment. Canvas course ID: " + canvasCourseId);
+            throw new AttendanceAssignmentException(AttendanceAssignmentException.Error.NON_EXISTENT_SECTION_ERROR);
         }
 
         AttendanceAssignment assignment = assignmentRepository.findByAttendanceSection(sections.get(0));
+        if(assignment == null || assignment.getAssignmentId() == null){
+            LOG.warn("No assignments found in assignment repository for section " + sections.get(0).getCanvasSectionId());
+            throw new AttendanceAssignmentException(AttendanceAssignmentException.Error.DELETION_ERROR);
+        }
 
         Optional<Assignment> canvasAssignment = Optional.empty();
         try{
             canvasAssignment = canvasApiWrapperService.getSingleAssignment(canvasCourseId, oauthToken, assignment.getCanvasAssignmentId().toString());
         } catch(IOException e){
-            LOG.warn("Assignment not found on canvas. " + e);
+            LOG.warn("Assignment " + assignment.getAssignmentId() + " not found on canvas for course " + canvasCourseId);
+            LOG.warn(e);
```

Comment:
Merge your descriptive message and your exception (stacktrace) into a single command.